lib/manager/composer/extract.ts

```diff
@@ -176,8 +181,11 @@ export async function extractPackageFile(
     return null;
   }
   res.deps = deps;
-  if (composerJson.type) {
-    res.managerData = { composerJsonType: composerJson.type };
+  if (is.string(composerJson.type)) {
+    const managerData: ComposerManagerData = {
+      composerJsonType: composerJson.type,
+    };
+    res.managerData = managerData;
```

Comment:
I can't leave a comment further up, but is it possible to change the return type of this method to `Promise<PackageFile<ComposerManagerData>>`?

I _think_ that should work, because the definition of `PackageFile` is `interface PackageFile<T = Record<string, any>> extends ManagerData<T>` and `ComposerManagerData` satisfies `Record<string, any>`

Here's a [TypeScript Playground link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLJxHA5tAInMOAHgBUA+ZAbwChk7kBbDbPAuALmROoF9rrQkWIhQBhAPYMADuIDO0dJhxR8hKrXoJJM+VABSs8SBIBPKRE6ywUUFgDcvfoOjwkyAAqIA1iwBiwABsIUmQAXmQAJQgtKAATIisbECwAGmQMEzIKCAAPSBBY2TRmZVVicio+AXAXEWKlaABBKWB1emRc60QwTwQfHH8ggAoASk53KElgeSJe-ohBlAAfZBAAVwCAsgcqhAC4WSKJaTkFEuhkYGkghghwIsUWKGbWmnbOqG65v0CIUfHJgxpsFvgNfkRjjozg0VGwKCt1psKG92sgoBAwGsoCAPIDgQA6dGGAIANz+iICIwc7T4PCAA) that shows a simplified example.